### PR TITLE
Avoid re-registering bslib components and globals

### LINKED ISSUE #1044: Using a `bslib::value_box()` breaks Quarto dashboards
This example from https://github.com/quarto-dev/quarto-cli/issues/9454

````markdown
---
format: dashboard
---

# Page 1

## {.sidebar}

This sidebar is not toggleable anymore

## Main body

### Value box

```{r}
bslib::value_box(
  'Adelie penguins',
  theme = "danger",
  value = 23432,
  full_screen = FALSE ## This is nevertheless possible in the dashboard 
)
```
````

breaks because this causes the bslib components js to be loaded, which attempts to register bslib's CustomElements that have already been registered by Quarto

```
localhost/:1 Uncaught DOMException: Failed to execute 'define' on 'CustomElementRegistry': the name "bslib-tooltip" has already been used with this registry
    at http://localhost:6054/issue-9454_files/libs/bslib-component-js-0.7.0/web-components.min.js:217:46
    at Array.forEach (<anonymous>)
    at http://localhost:6054/issue-9454_files/libs/bslib-component-js-0.7.0/web-components.min.js:217:19
    at http://localhost:6054/issue-9454_files/libs/bslib-component-js-0.7.0/web-components.min.js:217:183
```

I suspect the issue isn't just about `bslib-tooltip`, but also anything else where we've used a web component or custom element.

We probably need better coordination with Quarto on this one, but I think in the mean time we probably should avoid registering existing components.

### PR DESCRIPTION
**Pull request by gadenbuie**
Fixes #1044

Avoids redefining bslib-provided custom elements or web components (e.g. `bslib-tooltip`, `bslib-popover`) as well as globally-registered classes (`bslib.Sidebar` and `bslib.Card`).

In both cases a console error message is provided:

```
[bslib] Global window.bslib.Sidebar was already defined, using previous definition.
[bslib] Custom element bslib-tooltip was already defined, using previous definition.
```

### CODE REVIEW

File: srcts/src/components/sidebar.ts [hunk 1]
```diff
@@ -528,7 +528,5 @@ class SidebarInputBinding extends InputBinding {
 }
 
 registerBinding(SidebarInputBinding, "sidebar");
```
Comment (cpsievert) [inline srcts/src/components/sidebar.ts:530]: I'd be curious to know how `registerBinding()` handles this (i.e., does the last one win)? Seems like it'd be wise to match whatever it does?

Comment (gadenbuie) [reply srcts/src/components/sidebar.ts:530]: Would we want adding `bslib::value_box()` to a Quarto dashboard to use Quarto's components -- that they test against and expect to work -- or to use bslib's component js instead?

Comment (cpsievert) [reply srcts/src/components/sidebar.ts:530]: Ah, good point, makes sense for the 1st one to win then. But also, if the last one wins for `registerBinding()`, wouldn't the binding be using the wrong `Sidebar` (i.e., we should maybe update `registerBinding()` to do something similar to `registerBslibGlobal()`)?

Comment (gadenbuie) [reply srcts/src/components/sidebar.ts:530]: `registerBinding()` is basically a safe call to `BindingsRegistry.register()` in shiny. "registering" a binding just adds it to the end of an array. The `.getBindings()` method breaks ties using the `priority` propery, followed by most-recently-registered.

Anyway, you're pointing out that this isn't an unambiguous or complete fix, and I agree. This PR at least gets us out of the position of breaking Dashboards. We'll have to do some thinking in the future

**Review by cpsievert — APPROVED** [review]

**Pull request merged**